File: Python/Algorithms/1403.py
class Solution:
    def minSubsequence(self, nums):
        # Go through the list backwards.
        nums.sort(reverse=True)
        # Keep track of what's left, where we're at.
        left_sum, seq = sum(nums), 0

        # Continuously increase seq and decrease left_sum
        # until we reach point were seq > left_sum
        for i in range(len(nums)):
            v = nums[i]
            seq = seq + v
            left_sum = left_sum - v
            if (seq - left_sum) > 0:
                # done, return subsequence.
                return nums[:i+1]

File: Javascript/Algorithms/1403.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
let minSubsequence = nums => {
    let left_sum = nums.reduce((a, v) => a+=v), 
        seq = 0, 
        len = nums.length;
    nums.sort((a, b) => b - a);
    
    for (let i = 0; i < len; i++){
        let val = nums[i];
        left_sum -= val;
        seq += val;
        if ((seq - left_sum) > 0){
            return nums.slice(0, i+1);
        }
    }
};
